cmd/kaspaminer/client.go

```diff
@@ -5,42 +5,95 @@ import (
 	"github.com/kaspanet/kaspad/infrastructure/logger"
 	"github.com/kaspanet/kaspad/infrastructure/network/rpcclient"
 	"github.com/pkg/errors"
+	"sync"
+	"sync/atomic"
 	"time"
 )
 
 const minerTimeout = 10 * time.Second
 
 type minerClient struct {
-	*rpcclient.RPCClient
+	isReconnecting uint32
+	clientLock     sync.RWMutex
+	rpcClient      *rpcclient.RPCClient
 
+	cfg                        *configFlags
 	blockAddedNotificationChan chan struct{}
 }
 
-func newMinerClient(cfg *configFlags) (*minerClient, error) {
-	rpcAddress, err := cfg.NetParams().NormalizeRPCServerAddress(cfg.RPCServer)
-	if err != nil {
-		return nil, err
+func (mc *minerClient) safeRPCClient() *rpcclient.RPCClient {
+	mc.clientLock.RLock()
+	defer mc.clientLock.RUnlock()
+	return mc.rpcClient
+}
+
+func (mc *minerClient) reconnect() {
```

Comment:
Do we maybe want the reconnect logic on the rpcClient level (rather then minerClient)?

Reply:
Doing a general purpose reconnection logic is a far more complex task. Maybe we'll do it in the future

Reply:
Could you open a ticket for that?
IMHO this is pretty critical